src/wayland/shell/legacy/decoration.rs

```diff
@@ -0,0 +1,128 @@
+//! KDE Window decoration manager
+//!
+//! This interface allows a compositor to announce support for KDE's legacy server-side decorations.
+//!
+//! A client can use this protocol to request being decorated by a supporting compositor.
+//!
+//! ```no_run
+//! # extern crate wayland_server;
+//! #
+//! use smithay::wayland::shell::legacy::decoration::{init_kde_decoration_manager, KdeDecorationRequest};
+//! use wayland_protocols::misc::server_decoration::server::org_kde_kwin_server_decoration_manager::Mode;
+//!
+//! # let mut display = wayland_server::Display::new();
+//!
+//! let decoration_manager = init_kde_decoration_manager(
+//!     &mut display,
+//!     |decoration, request| match request {
+//!         KdeDecorationRequest::Setup { manager } => manager.default_mode(Mode::Server),
+//!         KdeDecorationRequest::NewToplevelDecoration { surface } => (),
+//!         KdeDecorationRequest::RequestMode { mode, .. } => decoration.mode(mode),
+//!         KdeDecorationRequest::Release => (),
+//!     },
+//!     None,
+//! );
+
+use std::{cell::RefCell, ops::Deref, rc::Rc};
+
+use wayland_protocols::misc::server_decoration::server::{
+    org_kde_kwin_server_decoration::{Mode, OrgKdeKwinServerDecoration, Request as ServerDecorationRequest},
+    org_kde_kwin_server_decoration_manager::{
+        OrgKdeKwinServerDecorationManager, Request as ServerDecorationManagerRequest,
+    },
+};
+use wayland_server::protocol::wl_surface::WlSurface;
+use wayland_server::{Display, Filter, Global, Main};
+
+/// Events generated by the kde decoration manager.
+#[derive(Debug)]
+pub enum KdeDecorationRequest {
+    /// Decoration manager setup completion.
+    Setup {
+        /// KDE decoration manager.
+        manager: OrgKdeKwinServerDecorationManager,
```

Comment:
off topic: KDE seems to have won the competition for most redundant interface names.